### PR TITLE
Base for having multiple secret accounts

### PR DESCRIPTION
**Pull request by enriquejosepadilla**
We need this to add another service account. The next one is necessary for getting logging out of the box.

### CODE REVIEW

File: django_cloud_deploy/workflow/_service_account.py [hunk 1]
```diff
@@ -30,26 +76,85 @@ def __init__(self, credentials: credentials.Credentials):
...
+            project_id: str) -> Dict[str, ServiceAccount]:
+        service_account_keys = {}
+        for key in self._service_account_workflow.load_service_accounts():
+            service_account_keys[key.id_] = key
+            self._service_account_workflow.create_key(project_id, key)
+
+        return service_account_keys
+
+    def create_key(self, project_id: str, service_account: ServiceAccount):
         """Enable required services for deploying Django apps to GKE.
```
Comment (wangxf123456) [inline django_cloud_deploy/workflow/_service_account.py]: This is my bad. Could you please help me fix this line?

File: django_cloud_deploy/workflow/_service_account.py [hunk 2]
```diff
@@ -16,12 +16,58 @@
...
+            output_path: The path to store your key.
+
+        Returns:
+            service_account: ServiceAccount
+        """
+        self.id_ = id_
+        self.name = name
+        self.file_name = file_name
+        self.roles = roles
+        if output_path == KEY_PATH_TEMPLATE:
```
Comment (wangxf123456) [inline django_cloud_deploy/workflow/_service_account.py]: output_path equals to a template is misleading. We can set the default value of output_path to None. Then we can write:

if not output_path:
    output_path =  KEY_PATH_TEMPLATE.format(os.path.expanduser('~'), id_)

File: django_cloud_deploy/workflow/_service_account.py [hunk 3]
```diff
@@ -16,12 +16,58 @@
...
+        roles: Roles the service account should have. Valid roles can be
+            found on https://cloud.google.com/iam/docs/understanding-roles
+        output_path: The path to store your key.
+    """
+    def __init__(self,
+                 id_: str,
+                 name: str,
+                 file_name: str,
+                 roles: List[str],
+                 output_path: str = KEY_PATH_TEMPLATE):
```
Comment (wangxf123456) [inline django_cloud_deploy/workflow/_service_account.py]: Lets rename it to "key_path".

File: django_cloud_deploy/workflow/_service_account.py (hunk 2 above)
Comment (enriquejosepadilla) [inline django_cloud_deploy/workflow/_service_account.py]: Use None.

Comment (enriquejosepadilla) [reply django_cloud_deploy/workflow/_service_account.py]: key_path

File: django_cloud_deploy/workflow/_service_account.py [hunk 4]
```diff
@@ -30,26 +76,85 @@ def __init__(self, credentials: credentials.Credentials):
...
-                   service_account_name: str, roles: List[str],
-                   output_path: str):
+    def handle_service_accounts(self, project_id: str,
+                                database_username: str,
+                                database_password: str):
+
+        services_accounts = self.create_service_accounts_and_keys(project_id)
+
+        # Prepare Kubernetes secret object based on service account keys.
+        secrets = self._service_account_workflow.generate_base_secrets(
```
Comment (enriquejosepadilla) [inline django_cloud_deploy/workflow/_service_account.py]: remove _service_account_workflow. copy/paste bug.

File: django_cloud_deploy/workflow/_service_account.py [hunk 5]
```diff
@@ -30,26 +76,85 @@ def __init__(self, credentials: credentials.Credentials):
...
+                key_content = key_file.read()
+
+            secrets[key_id] = {
+                key.file_name: key_content
+            }
+
+        return secrets
+
+    @staticmethod
+    def load_service_accounts():
```
Comment (enriquejosepadilla) [inline django_cloud_deploy/workflow/_service_account.py]: add docs.

File: django_cloud_deploy/workflow/_service_account.py [hunk 6]
```diff
@@ -30,26 +76,85 @@ def __init__(self, credentials: credentials.Credentials):
...
+        self._service_account_workflow.generate_service_account_secrets(
+            services_accounts, secrets
+        )
+        return secrets
+
+    def create_service_accounts_and_keys(
+            self,
+            project_id: str) -> Dict[str, ServiceAccount]:
+        service_account_keys = {}
+        for key in self._service_account_workflow.load_service_accounts():
```
Comment (enriquejosepadilla) [inline django_cloud_deploy/workflow/_service_account.py]: rename.

File: django_cloud_deploy/workflow/_service_account.py [hunk 7]
```diff
@@ -30,26 +76,85 @@ def __init__(self, credentials: credentials.Credentials):
...
+            project_id, service_account.id_)
+        with open(service_account.output_path, 'w') as output_file:
             output_file.write(json.dumps(key_data))
+
+    @staticmethod
+    def generate_service_account_secrets(
+            service_account_keys: Dict[str, ServiceAccount],
+            secrets: Dict[str, Any] = {}):
+
+        for key_id, key in service_account_keys:
```
Comment (enriquejosepadilla) [inline django_cloud_deploy/workflow/_service_account.py]: .items()

**Pull request merged**